lib/Service/AccountService.php

```diff
@@ -78,6 +78,46 @@ public function findById(int $id): Account {
 		return new Account($this->mapper->findById($id));
 	}
 
+	/**
+	 * Finds a mail account by user id and mail address
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $userId			system user id
+	 * @param string $address			mail address (e.g. test@example.com)
+	 *
+	 * @return Account[]
+	 *
+	 * @throws ClientException
+	 */
+	public function findByUserIdAndAddress(string $userId, string $address): array {
+		// evaluate if cached accounts collection already exists
+		if (isset($this->accounts[$userId])) {
+			// initialize tempory collection
```

Comment:
```suggestion
			// initialize temporary collection
```

The comment that first cached accounts are evaluate is helpful, that we are initializing a temporary collection here, iterate, match, and early exit are not. 

Good read: https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/